distributed/protocol/cupy.py

```diff
@@ -6,21 +6,15 @@
 
 
 class PatchedCudaArrayInterface:
-    """This class do two things:
-        1) Makes sure that __cuda_array_interface__['strides']
-           behaves as specified in the protocol.
-        2) Makes sure that the cuda context is active
+    """This class does one thing:
+        1) Makes sure that the cuda context is active
            when deallocating the base cuda array.
         Notice, this is only needed when the array to deserialize
         isn't a native cupy array.
     """
 
     def __init__(self, ary):
-        cai = ary.__cuda_array_interface__
-        cai_cupy_vsn = cupy.ndarray(0).__cuda_array_interface__["version"]
-        if cai.get("strides") is None and cai_cupy_vsn < 2:
```

Comment:
My memory of this is fuzzy - Is there no longer a chance the cupy/numba combination can be a problem here?

Reply:
Good question. In this PR are we are now always storing the actual `strides` in the `header`. So no it's not possible.